apps/explorer/src/ui/CoinBalance.tsx

```diff
@@ -1,21 +1,38 @@
 // Copyright (c) Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { Heading } from '~/ui/Heading';
-import { Text } from '~/ui/Text';
+import {
+    useFormatCoin,
+    CoinFormat,
+    formatBalance,
+} from '~/hooks/useFormatCoin';
+import { Amount } from '~/ui/Amount';
 
-// Passing amount as a string or number for optional number suffix
-export type CoinBalanceProps = { amount: number | string; symbol?: string };
+export interface CoinBalanceProps {
+    amount: number | string | bigint;
+    symbol?: string | null;
+    size?: 'lg' | 'md';
+    coinFormat?: keyof typeof CoinFormat;
+}
+
+const DECIMALS = 0;
 
-export function CoinBalance({ amount, symbol }: CoinBalanceProps) {
-    return (
-        <div className="flex items-end gap-1 text-sui-grey-100 break-words">
-            <Heading variant="heading4">{amount}</Heading>
-            {symbol && (
-                <div className="text-sui-grey-80">
-                    <Text variant="bodySmall">{symbol}</Text>
-                </div>
-            )}
-        </div>
+// Passing amount as a string or number for optional number suffix
+export function CoinBalance({
+    amount,
+    symbol,
+    size = 'md',
```

Comment:
Leave defaulting of this and the coinFormat to the `Amount` component, this component should just wrap it.